### ISSUE TITLE
btrfs.go args.lim.max_referenced undefined

### ISSUE THREAD

**Issue by siretart** [containers/storage#1454]
Compiling against newer https://github.com/containerd/btrfs (version 1.0) results in a compilation failure:

```
# github.com/containers/storage/drivers/btrfs
src/github.com/containers/storage/drivers/btrfs/btrfs.go:385:11: args.lim.max_referenced undefined (type _Ctype_struct_btrfs_qgroup_limit has no field or method max_referenced)
```

This was reported in debian as https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1027116 and https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1027116

**Comment by siretart** [comment]
This issue is very similar to https://github.com/moby/moby/issues/44698 and seems to be triggered by an upstream change referenced here: https://github.com/kdave/btrfs-progs/commit/03451430de7cd2fad18b0f01745545758bc975a5

**Comment by siretart** [comment]
in order to get this working in debian, the following code-change appears to do it: https://salsa.debian.org/go-team/packages/golang-github-containers-storage/-/blob/342826669b5a5317db2342abbfe94b1e96c8c460/debian/patches/btrfs-1.0.patch

**Comment by flouthoc** [comment]
Crun CI is blocked (https://github.com/containers/crun/pull/1111) for similar reason I have created a PR.

**Comment by kdave** [comment]
The change in btrfs-progs will be reverted and released in v6.1.1 so you'll probably see more build breakage until everybody updates.
